timemachine/fe/free_energy.py

```diff
@@ -1162,7 +1202,7 @@ def get_equilibrated_xvb(xvb: CoordsVelBox, state_idx: StateIdx) -> CoordsVelBox
 
     for current_frame in range(md_params.n_frames):
 
-        def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> Trajectory:
+        def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> Tuple[NDArray, NDArray, NDArray, Optional[float]]:
```

Comment:
nit: wonder if it'd be worth defining `Sample = Tuple[NDArray, NDArray, NDArray, Optional[float]]` for clarity?

Reply:
Its an easy thing to do, but not clear what the value would be. Mostly because this type has changed to meet our needs over time, returning the final velocities and the state of the barostat, which feels too ill-defined to warrant a type hint devoted to do. 

Also there are already several type hints for 'Sample[s]' type aliases that are slightly different which makes me further hesitant to add another type without trying to unify this:
https://github.com/proteneer/timemachine/blob/master/timemachine/md/hrex.py#L129
https://github.com/proteneer/timemachine/blob/master/timemachine/md/smc.py#L13-L14
https://github.com/proteneer/timemachine/blob/master/timemachine/fe/reweighting.py#L16

Reply:
Fair enough; only if you think it improves overall clarity. Re multiple instances of type aliases with the same name: I think it's perfectly OK to have local type aliases with the same name as a type alias in another module (kind of seems like a quirk of Python that type aliases have to be top-level value bindings), but we might want to start prefixing aliases intended to by local with an underscore or similar.